Zend/Optimizer/zend_ssa.c

```diff
@@ -794,7 +794,23 @@ static zend_always_inline int _zend_ssa_rename_op(const zend_op_array *op_array,
 					}
 				}
 			}
+			break;
 		}
+		case ZEND_FETCH_OBJ_R:
+		case ZEND_FETCH_OBJ_IS:
+		case ZEND_FETCH_OBJ_RW:
+		case ZEND_FETCH_OBJ_W:
+		case ZEND_FETCH_OBJ_UNSET:
+		case ZEND_FETCH_OBJ_FUNC_ARG:
+			if ((build_flags & ZEND_SSA_RC_INFERENCE) && (opline->op1_type & (IS_CV|IS_VAR|IS_TMP_VAR))) {
+				goto add_op1_def;
+			}
+			break;
+		case ZEND_INIT_METHOD_CALL:
+			if ((build_flags & ZEND_SSA_RC_INFERENCE) && (opline->op1_type & (IS_CV|IS_VAR|IS_TMP_VAR))) {
+				goto add_op1_def;
+			}
+			break;
```

Comment:
After this, I'm not sure if it makes sense to relay on IS_OBJECT RC inference at all.
Almost any operation makes objects  ``RC1|RCN`` anyway.

Reply:
Ok. In what way would you prefer fixing this? Adding a check to `UPDATE_SSA_TYPE()`? This may prevent RC-inference from being utilized for pi variables, but might not make a big difference in practice.

Reply:
Adding new SSA variables for IS_VAR/IS_TMP_VAR doesn't make any sense, because they are not goinf to be used (IS_VAR/IS_TMP_VAR are used once).